### DISCUSSION TITLE
Issue with Metamask connection and modal control using React Native

### DISCUSSION THREAD

**Discussion by ShuraRB** [WalletConnect/walletconnect-monorepo#2597] [Developer Support] [answered]
Hi... I'm using the hook to control the modal with my custom component. The modal opens correctly, but when I select Metamask and go to the Metamask app, nothing happens. There's no option to connect with my React Native app, and when I go back to my app, there isn't a connection with my wallet. this also happened when I use` <Web3Button />.`

i open the modal with this component

`<LogoButton svgElement={<WalletCSvg height={50} width={50} />} onPress={open} />`

my provider metadata is like the example 

`export const providerMetadata: IProviderMetadata = {
  name: 'React Native V2 dApp',
  description: 'RN dApp by WalletConnect',
  url: 'https://walletconnect.com/',
  icons: ['https://avatars.githubusercontent.com/u/37784886'],
  redirect: {
    native: 'w3msample://',
  },
};
`
Obviously with my app info, that is almost same like the example

**Comment by ignaciosantise** [comment, marked as answer]
Hey @ShuraRB 👋  MetaMask will release a new version that supports WalletConnect V2 soon. As you can see [here](https://github.com/MetaMask/metamask-mobile/pull/6380) they already merged the implementation code. Until they release this version, you won't be able to connect with MetaMask.
